### PR TITLE
Rename usage of apply and undo for transaction - Closes #2198

### LINKED ISSUE #2198: Rename usages of .apply from the application.
### Expected behavior
The application should not override prototype properties defined in javascript.
### Actual behavior
In many parts of the application i.e. transactions and blocks, we use the function name `apply` to perform some database writes. Which means we are overriding javascript defined `apply` function. Which makes code
a- harder to understand
b- restricts usage of `apply` function in the future.
c- is not a best practice.
### Which version(s) does this affect? (Environment, OS, etc...)
1.1.0

### PR DESCRIPTION
**Pull request by SargeKhan**
### What was the problem?
apply was is a javascript defined property for functions, we were using the same variable name for transaction application.
### How did I fix it?
Changed apply to applyConfirmed and for consistency renamed undo to undoConfirmed.
### How to test it?
Run tests.
### Review checklist

* The PR solves #2198 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by SargeKhan** [comment]
This PR should be merged after #2195

File: logic/transaction.js [hunk 1]
```diff
@@ -916,7 +916,7 @@ class Transaction {
 					return setImmediate(cb, mergeErr);
 				}
 
-				__private.types[transaction.type].undo(
+				__private.types[transaction.type].undoConfirmed(
 					transaction,
 					block,
 					sender,
```
Comment (4miners) [inline logic/transaction.js]: `undoErr` -> `undoConfimedErr`

File: modules/blocks/chain.js [hunk 2]
```diff
@@ -585,21 +585,24 @@ __private.undoStep = function(transaction, oldLastBlock, tx) {
...
 					return setImmediate(reject, accountErr);
 				}
 				// Undoing confirmed transaction - refresh confirmed balance (see: logic.transaction.undo, logic.transfer.undo)
 				// WARNING: DB_WRITE
-				modules.transactions.undo(
+				modules.transactions.undoConfirmed(
 					transaction,
 					oldLastBlock,
 					sender,
 					undoErr => {
```
Comment (4miners) [inline modules/blocks/chain.js]: `undoConfirmedErr`

**Comment by SargeKhan** [comment]
@MaciejBaj the tests are failing because one hotfix was added in 1.1.0, and 1.1.0 is not merged into 1.2.0

File: modules/blocks/chain.js [hunk 3]
```diff
@@ -389,22 +389,22 @@ __private.applyConfirmedStep = function(block, tx) {
 					{ publicKey: transaction.senderPublicKey },
 					(accountErr, sender) => {
 						if (accountErr) {
-							const err = `Failed to get account to apply transaction: ${
+							const err = `Failed to get account to applyConfirmed transaction: ${
```
Comment (nazarhussain) [inline modules/blocks/chain.js]: For consistency may be good, but semantically this statement does not feel good to me. 

> Failed to get account to applyConfirmed transaction

May be instead

> Failed to get account to apply transaction confirmed state. 

function naming convention are internal stuff, while log messages are what our app distribute outside as information. So it should be semantically and grammatically correct as well.

File: modules/blocks/chain.js [hunk 4]
```diff
@@ -389,22 +389,22 @@ __private.applyConfirmedStep = function(block, tx) {
...
-						modules.transactions.apply(
+						modules.transactions.applyConfirmed(
 							transaction,
 							block,
 							sender,
 							err => {
 								if (err) {
 									// Fatal error, memory tables will be inconsistent
-									err = `Failed to apply transaction: ${
+									err = `Failed to applyConfirmed transaction: ${
```
Comment (nazarhussain) [inline modules/blocks/chain.js]: Same here.

File: modules/blocks/chain.js [hunk 5]
```diff
@@ -585,22 +585,25 @@ __private.undoStep = function(transaction, oldLastBlock, tx) {
 				if (accountErr) {
 					// Fatal error, memory tables will be inconsistent
 					library.logger.error(
-						'Failed to get account to undo transactions',
+						'Failed to get account to undoConfirmed transactions',
```
Comment (nazarhussain) [inline modules/blocks/chain.js]: Same here. 
> Failed to get account to undo confirmed state of transactions

File: modules/blocks/chain.js [hunk 6]
```diff
@@ -624,11 +627,14 @@ __private.undoUnconfirmStep = function(transaction, tx) {
...
 			transaction,
-			undoUnconfirmErr => {
-				if (undoUnconfirmErr) {
+			undoUnconfirmedErr => {
+				if (undoUnconfirmedErr) {
 					// Fatal error, memory tables will be inconsistent
-					library.logger.error('Failed to undo transactions', undoUnconfirmErr);
-					return setImmediate(reject, undoUnconfirmErr);
+					library.logger.error(
+						'Failed to undoUnconfirmed transactions',
```
Comment (nazarhussain) [inline modules/blocks/chain.js]: > Failed to undo unconfirmed state of transactions

**Pull request merged**